### ISSUE TITLE
Add `SCREAMING_SNAKE_CASE` to `key-format-style` rule

### ISSUE THREAD

**Issue by nolde** [intlify/eslint-plugin-vue-i18n#137]
**What rule do you want to change?**
`@intlify/vue-i18n/key-format-style`

**Does this change cause the rule to produce more or fewer warnings?**
Depends.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option, slightly changed behaviour (`snake_case` would stop matching `SCREAMING_SNAKE_CASE`).

**Please provide some example code that this change will affect:**
Allow `SCREAMING_SNAKE_CASE` as an allowed option.

```vue
<i18n>
{
  "en": {
    "states": {
      "RUNNING": "Your process is still running.",
      "COMPLETED": "The process has been completed successfully.",
      "FAILED": "The process has failed."
    }
  }
}
</i18n>
```

**What does the rule currently do for this code?**
There is no option that differentiates `SCREAMING_SNAKE_CASE` from common `snake_case` right now.

**What will the rule do after it's changed?**
`SCREAMING_SNAKE_CASE` naming pattern is pretty common for enumerations and other similar constants in JavaScript. By making it _different_ from normal `snake_case` and allowing independent setting, this pattern can be supported.

**Additional context**
[Google's JavaScript naming conventions](https://google.github.io/styleguide/javascriptguide.xml?showone=Naming#Naming)
